modules/protobuf/src/main/scala/smithy4s/protobuf/internals/Wire.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.protobuf.internals
+
+private[internals] object Wire {
+
+  def encodeTag(protoIndex: Int, wireType: Int): Int =
+    (protoIndex << 3) | wireType
+
+  def decodeTag(byte: Byte): (Int, Int) = {
+    val m = byte & 0xff
+    (m >> 3, m & 0x7)
+  }
+
+  // See https://protobuf.dev/programming-guides/encoding/#structure
+  //
+  // NB : not ascribing types to get special encoding in bytecode.
```

Comment:
**question:** (optional) - I'd be interested in the particular encoding we're aiming for. Also, would `final val` be of any help here? (hoping for inlining)

Reply:
I think I was going for inlined constant, but it's been ages and you're absolutely right that I'm missing the final

Reply:
[9334fe5](https://github.com/disneystreaming/smithy4s/pull/1455/commits/9334fe568a80d34a2b356cb3a4615ea26a0b6fb5)